src/addon/cc-addon-option-form.js

```diff
@@ -0,0 +1,124 @@
+import '../addon/cc-addon-option.js';
+import '../atoms/cc-button.js';
+import '../molecules/cc-block.js';
+import { css, html, LitElement } from 'lit-element';
+import { dispatchCustomEvent } from '../lib/events.js';
+import { i18n } from '../lib/i18n.js';
+
+/**
+ * A component that displays a form of `<cc-addon-option>`.
+ *
+ * * 🎨 default CSS display: `block`
+ * <br>
+ * 🧐 [component's source code on GitHub](https://github.com/CleverCloud/clever-components/blob/master/src/addon/cc-addon-option-form.js)
+ *
+ * ## Details
+ *
+ * * Emits a `submit` event when the form is submited. The emitted event is an object of options state (enabled / disabled).
+ *
+ * ## Type definitions
+ *
+ * ```js
+ * interface Option {
+ *   title: String,
+ *   logo: String,
+ *   description: DOM,
+ *   name: String,
+ *   enabled: Boolean
+ * }
+ *
+ * interface Options {
+ *   optionName1: Boolean,
+ *   optionName2: Boolean,
+ * }
+ * ```
+ *
+ * @prop {String} title - Title of the whole options form.
+ * @prop {Array<Option>} options - List of Option object to render.
+ *
+ * @event {CustomEvent<Options>} cc-addon-option-form:submit - Fires when the form is submitted.
+ *
+ * @slot description - The description of the add-on and available options.
+ */
+export class CcAddonOptionForm extends LitElement {
+
+  static get properties () {
+    return {
+      title: { type: String, attribute: 'title' },
+      options: { type: Array, attribute: 'options' },
+    };
+  }
+
+  constructor () {
+    super();
+    this.title = null;
+    this.options = [];
+    this._optionsStates = {};
+  }
+
+  _onSubmit () {
+    // If some options were not changed, fill them here
+    this.options.forEach((option) => {
+      if (this._optionsStates[option.name] === undefined) {
+        this._optionsStates[option.name] = option.enabled || false;
+      }
+    });
+
+    dispatchCustomEvent(this, 'submit', this._optionsStates);
+  }
+
+  _onOptionToggle (optionName) {
+    return ({ detail }) => {
+      this._optionsStates[optionName] = detail;
+    };
+  }
```

Comment:
In other components we don't return an arrow function in an event handler, we pass the event object directly in the template.

The handler would look like this:

```js
_onOptionToggle ({ detail }, optionName) {
  this._optionsStates[optionName] = detail;
}
```

and in the template, you do this:

```
@cc-addon-option:input=${(e) => this._onOptionToggle(e, option.name)}
```